### ISSUE TITLE
Lint: no downloads in build.sh

### ISSUE THREAD

**Issue by epruesse** [bioconda/bioconda-utils#469]
No use of
- `git clone`
- `git submodule`
- `wget`
- `curl`

"All sources must be specified via source: url:`"

**Comment by dpryan79** [comment]
Am I interpreting this correctly in that you'd propose forbidding using `git`/`wget`/`curl` in `build.sh`?

**Comment by epruesse** [comment]
Yes. I guess we need to discuss this. It could be a warning I suppose.

My arguments would be:
 - if we allow downloading arbitrary sources in the `build.sh`, we might as well ditch the `checksum` as it doesn't cover all sources anyway
 - at least GPL requires us to provide the sources
 - cargo-port archival doesn't catch these downloads

I'm open to pragmatic solutions, but personally I do believe that we shouldn't have downloads in the `build.sh`.

**Comment by epruesse** [comment]
Meh. `gappa` uses `cmake` to download stuff during build. That is really hard to catch.

The problem is this: GPL requires that **we** must ensure that anyone who gets the binaries from us can build the tool. So **we** must be able to give them the entire source code. It is not sufficient to rely on external sources, we have to provide a copy ourselves. That's just the way the license is, nothing to be done about it.

**Comment by epruesse** [comment]
I think we need to have the build run in a container without internet access. That would fix all of this in one fell swoop.

**Comment by dpryan79** [comment]
I think we're just going to ignore that the GPL requires this.

**Comment by daler** [comment]
How do conda-forge and anaconda handle this?

**Comment by epruesse** [comment]
I don't think they have established procedures. With `cargo-port`, we are already ahead.

**Comment by bgruening** [comment]
I'm with @dpryan79 here. Let's "postpone" this.
